tests/utils.py

```diff
@@ -9,8 +12,65 @@ class Keys(str, Enum):
 KEY_LOCK = Keys.KEY + "-lock"
 
 
-def ensure_key(key):
+def ensure_key(key: Union[str, Enum]) -> str:
     if isinstance(key, Enum):
         return key.value
     else:
         return key
+
+
+class AbstractBaseCache(BaseCache[str]):
+    """BaseCache that can be mocked for NotImplementedError tests"""
```

Comment:
I feel like these 2 classes are rather awkward, but I'll merge it for now.

I don't think there's any value in testing the logic of abstract methods, and I'm wondering if the other tests would still be testing the same thing if we just use the memory cache instead..

Reply:
Also introduces a new mypy error:
https://github.com/aio-libs/aiocache/actions/runs/4333749520/jobs/7567135931

Curiously, only on the first method, though it looks like it should happen on all of them. Maybe because it doesn't have an await...

Reply:
> Also introduces a new mypy error: https://github.com/aio-libs/aiocache/actions/runs/4333749520/jobs/7567135931

Because this class exists purely for tests, to allow mocks without throwing ABC `TypeError`s, I think adding `# type: ignore[safe-super]` would be appropriate here.

Reply:
I realise that, it's just a side note to the first comment about whether we can just remove them if they are not testing anything useful.

Reply:
> I feel like these 2 classes are rather awkward, but I'll merge it for now.
> 

Yeah, it's a bit awkward because the tests aren't being performed **_directly_** on `BaseCache`, and because of how many abstract methods need to be defined on the derived class.  However the definitions of `AbstractBaseCache` and `ConcreteBaseCache` are trivial and transparent, so at least no new complexity is added.


> I don't think there's any value in testing the logic of abstract methods, and I'm wondering if the other tests would still be testing the same thing if we just use the memory cache instead.. If so, we could then drop both of these classes.

On the contrary, I really like how the logic of what's defined in `BaseCache` gets tested independently with the (abstract) base class.  

The derived classes used for the backend implementations use more resources and add enough "side effects" that proper testing of the underlying logic would require repeating these tests with all 3 backends.  And even then, it would perhaps be an open question of whether the `BaseCache` functionality extends to new backends.

Keeping these tests of `BaseCache` separate from tests of the backends seems much cleaner and more robust. Previously this was a bit simpler because `BaseCache` was not formally abstract--it had unimplemented methods intended to be override, but instantiating `BaseCache` did not raise an exception.

Well, in any case, I'm open to finding a cleaner test solution, but I favor keeping the `BaseCache` tests separate from the backend tests of the same methods.